Multiple changes

p0psicles opened the pull request:
Changes listed in the comments

I still need to double check I didn't leave anything out. Have to compare to the branch i've been working on.

moogar0880 commented on trakt/tv.py:
```diff
@@ -13,8 +13,9 @@
 __all__ = ['dismiss_recommendation', 'get_recommended_shows', 'genres',
-           'popular_shows', 'trending_shows', 'updated_shows', 'TVShow',
-           'TVEpisode', 'TVSeason', 'Translation']
+           'popular_shows', 'trending_shows', 'updated_shows', 'recommmended_shows', 
```
`recommmended_shows` → `recommended_shows`

moogar0880 commented:
The linter seems unhappy about some trailing white space and line lengths:

```
trakt/sync.py:12:80: E501 line too long (82 > 79 characters)
trakt/sync.py:79:80: E501 line too long (81 > 79 characters)
trakt/sync.py:97:80: E501 line too long (81 > 79 characters)
trakt/sync.py:114:80: E501 line too long (81 > 79 characters)
trakt/sync.py:132:80: E501 line too long (81 > 79 characters)
trakt/sync.py:149:80: E501 line too long (81 > 79 characters)
trakt/tv.py:16:80: E501 line too long (84 > 79 characters)
trakt/tv.py:16:85: W291 trailing whitespace
trakt/tv.py:17:80: E501 line too long (83 > 79 characters)
trakt/tv.py:96:80: E501 line too long (81 > 79 characters)
trakt/tv.py:100:80: E501 line too long (83 > 79 characters)
trakt/tv.py:115:80: E501 line too long (98 > 79 characters)
trakt/tv.py:116:80: E501 line too long (80 > 79 characters)
trakt/tv.py:119:80: E501 line too long (83 > 79 characters)
trakt/tv.py:133:80: E501 line too long (92 > 79 characters)
trakt/tv.py:137:80: E501 line too long (83 > 79 characters)
trakt/tv.py:151:80: E501 line too long (94 > 79 characters)
trakt/tv.py:155:80: E501 line too long (83 > 79 characters)
trakt/tv.py:169:80: E501 line too long (84 > 79 characters)
trakt/tv.py:170:80: E501 line too long (86 > 79 characters)
```

Otherwise these changes looks good to me so far

p0psicles commented:
When I made my changes, it was using my own linter config. Ill fix those

p0psicles commented:
@moogar0880 i've added a number of return's for functions that call a @post route.
I've also fixed the tests for these. I've made use of pytest parametrizing for this. Like in:
https://github.com/moogar0880/PyTrakt/pull/133/commits/743814d480614aa3541a74e728b5b4855272d273#diff-7cdc82c5ea3109d2377741f06aecdcce8d49257482335f49ee25be99853621b3R61

This is a better way of utilizing mulitple asserts in one test method.

I'll also add tests for the new post routes.

p0psicles commented:
@moogar0880 doesn't deserve the most beautifull coding price. But it works.
https://github.com/moogar0880/PyTrakt/pull/133/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R31

If I don't use open(,encoding='utf-8') it fails some tests because it can't open movies.json. I suspect it's because the default encoding on windows, is not utf-8

p0psicles commented:
@moogar0880 thought I could flesh out the individual changes into separate PR. If you didn't like some of them. But if this PR is okay. What you think of just working on this PR. And merge that one when finished?

Other question. How do you run the `make` commands on a windows machine?

moogar0880 commented:
> But if this PR is okay. What you think of just working on this PR. And merge that one when finished?

I think keeping things in one PR is fine for now.

> Other question. How do you run the make commands on a windows machine?

That's a good question, I don't have a windows machine to test on so I can't really say for sure. I suspect things will more or less Just Work™ if you have WSL available. If not, then it looks like there is a [GnuWin32](http://gnuwin32.sourceforge.net/packages/make.htm) project you could try, although I unfortunately don't have much in the way of guidance on that end.

p0psicles commented:
@moogar0880 would you be so kind to give me a review on the PR. Then I know where I need to further focus on.
In the mean time i'm setting up ubuntu to run the build. I'll look into how to add to the docs.

p0psicles commented:
https://www.sphinx-doc.org/en/master/usage/advanced/setuptools.html

Ever considered using setuptools? Would make it cross platform buildable

p0psicles commented:
@moogar0880 I got the hang of sphinx now a little. And I have to say. It's pretty awesome.
But I really need to have a discussion with you now, before I start to make all kinds of changes like for example to the way we authenticate, set trakt config location, set auth method, etc.

I've started with adding docs/sync.rst, as I noticed it was missing.

moogar0880 commented on trakt/sync.py:
```diff
@@ -65,57 +68,100 @@ def add_to_history(media, watched_at=None):
 
-    :param media: The :class:`TVShow` to remove from your watchlist
+@post
+def remove_from_watchlist(media=None):
```
is `None` actually a valid default value here? I don't see anything in the trakt API docs about supporting null request bodies for this endpoint, which I think is what would happen if someone were to call `remove_from_watchlist()` with no argument.

moogar0880 reviewed (commented):
Left a comment about one of the function signature changes, but otherwise I think this is looking really good.

> Ever considered using setuptools? Would make it cross platform buildable

I wasn't aware that spinx had that integration, that definitely looks like it would worth looking into.

> I've started with adding docs/sync.rst, as I noticed it was missing.

Awesome! Thanks so much for doing that.

> But I really need to have a discussion with you now, before I start to make all kinds of changes like for example to the way we authenticate, set trakt config location, set auth method, etc.

For sure. What kinds of changes are you thinking for those? I also have a couple of things I'd like to change if we're going to be making a bunch of breaking changes, most notably dropping support for Python 2, but I'm definitely interested to hear what changes you're looking into as well

p0psicles commented on trakt/core.py:
```diff
@@ -19,9 +19,10 @@
+           'init', 'BASE_URL', 'CLIENT_ID', 'CLIENT_SECRET', 'DEVICE_AUTH',
+           'REDIRECT_URI', 'HEADERS', 'CONFIG_PATH', 'OAUTH_TOKEN',
+           'OAUTH_REFRESH', 'PIN_AUTH', 'OAUTH_AUTH', 'AUTH_METHOD',
+           'APPLICATION_ID', 'get_device_code']
```
WaWaddle load

p0psicles commented on trakt/calendar.py:
```diff
@@ -60,14 +66,23 @@ def _get(self):
+            e_data = {
+                'airs_at': airs_date(first_aired),
+                'ids': episode.get('ids'),
+                'title': show_data.get('title'),
```
This might be considered a breaking change. But pytrakt was leaving out TV show information. And I really need this.

moogar0880 replied:
Interesting... Does this actually do what you want it to do? It seems like this would result in a `TVEpsiode` with a title of the show and not the episode

moogar0880 replied:
Should this line be `'title': episode.get('title')` since this is building up the dict for the `TVEpisode`?

p0psicles replied:
Should I remove it? it doesn't have a purpose for my usecase

p0psicles commented on trakt/calendar.py:
```diff
@@ -60,14 +66,23 @@ def _get(self):
+                'airs_at': airs_date(first_aired),
+                'ids': episode.get('ids'),
+                'title': show_data.get('title'),
+                'show_data': TVShow(**show_data)
```
Tagged the wrong line. I put the show object in show_data. But mostly in the app show property has the show title.

moogar0880 replied:
I think this line should be fine

p0psicles commented:
DId you still wanted to make changes in this branch? Or are we good to go? I can give you access to my fork if you want.

moogar0880 commented:
@p0psicles I think we can merge this as-is and I can make my changes in a follow up PR. I'll get this up into PyPi as `2.15.0` today.

Thanks for all your work on this!
